Revisited initial Bridge parameters

pdyraga opened the pull request:
Closes #258 

Parameters are set to be more forgiving for the early days of the network. Timeouts are pretty long and slashing amounts are minimal.

During this work, I realized we should improve validation for deposit and redemption parameters. The deposit dust threshold must be greater than the per-deposit max allowed Bitcoin network fee. The redemption dust threshold must be greater than the per-redemption max allowed Bitcoin network fee. Added the validation in f9f75faeed956fbce381be8d11017a23097877c1.

beaushinkle commented:
Compiling the changes in english:

* all max mining fees: `0.0001 BTC => 0.001 BTC`
* all slashing punishments: `10_000 T => 100 T`
* redemption timeout: `48 hours => 5 days`
* failed redemption slashing amount: `10_000 T => 100 T`
* moving funds dust amount: `0.0002 BTC => 0.002 BTC `
* failed moving funds slashing amount: `10_000 T => 100 T`
* fraud challenge deposit amount: `2 eth => 5 eth`
* wallet btc transfer max: `10 btc => 50 btc`

Does that look accurate / was that the intent? 

Bitcoin mining fee looks good. Here's a chart of the last 3 years:

<img width="1025" alt="image" src="https://user-images.githubusercontent.com/1045160/170803153-813f8ed3-fab8-4f36-8091-d41cad314714.png">

Other than that weird peak at $63/tx on 2021-04-21, fees have been *comfortably* below 30$.

In the early days, we want to be really light on punishments, and slashing 100T certainly accomplishes that, and making folks pay 5 eth to challenge does too.

Same idea for giving operators 5 days to redeem - if something is janky this gives us time to release a new client version to try to help them fix their thing.

In this system, I would actually want to keep the `walletMaxBtcTransfer` low; since we don't want BTC to pile up *at all* (since the btc isn't protected by punishments), so I would keep it at 10 rather than 50. If we don't have enough wallets, what ends up happening is that we just divide evenly among all the live wallets anyway :)

beaushinkle commented on solidity/contracts/bridge/BridgeState.sol:
```diff
@@ -369,6 +370,11 @@ library BridgeState {
+        require(
+            _depositDustThreshold > _depositTxMaxFee,
+            "Deposit dust threshold must be greater than deposit TX max fee"
+        );
```
very reasonable

pdyraga commented:
> Compiling the changes in english:
> 
> * all max mining fees: `0.0001 BTC => 0.001 BTC`
> * all slashing punishments: `10_000 T => 100 T`
> * redemption timeout: `48 hours => 5 days`
> * failed redemption slashing amount: `10_000 T => 100 T`
> * moving funds dust amount: `0.0002 BTC => 0.002 BTC `
> * failed moving funds slashing amount: `10_000 T => 100 T`
> * fraud challenge deposit amount: `2 eth => 5 eth`
> * wallet btc transfer max: `10 btc => 50 btc`
> 
> Does that look accurate / was that the intent?

Yes, except that I changed `walletMaxBtcTransfer` back to 10 BTC according to your suggestion.

Slashing amounts are low to do not punish early stakers too much in case there are bugs in the client.

All max mining fees are pretty high - while for most of the time we should have wallet executing transactions using fees not even close to the max ones, I think leaving some margin makes sense, especially early days when we'll have more to worry about.

All timeouts are pretty long and I think they should stay that long. One of the lessons from v1 is that 6 hours timeouts were just too much and operators located in different time zones could not even start coordinating problem resolution before the timeout was hit. This should not be that problematic with 51-of-100 threshold, but I think having enough time margin is important especially early days.

> In this system, I would actually want to keep the `walletMaxBtcTransfer` low; since we don't want BTC to pile up _at all_ (since the btc isn't protected by punishments), so I would keep it at 10 rather than 50. If we don't have enough wallets, what ends up happening is that we divide evenly among all the live wallets anyway :)

You are right, changed back in https://github.com/keep-network/tbtc-v2/pull/352/commits/12c9bf665103d2d2641c568a2dbafc6b606d37ab.
